mlir/utils/jenkins/parameterSweeps.py

```diff
@@ -0,0 +1,403 @@
+Output = {runnerOut}
+Errors = {runnerErrs.decode('utf-8')}""", file=sys.stderr)
+        return TestResult.FAIL
+    return TestResult.PASS
+
+IterType = TypeVar('IterType')
+def grouper(iterable: Iterable[IterType], n: int):
+    it = iter(iterable)
+    while True:
+        chunk = tuple(itertools.islice(it, n))
+        if not chunk:
+            return
+        yield chunk
+
+async def dropGoodConfig(config: ConvConfiguration,
+        options: Options) -> Union[TestResult, ConvConfiguration]:
+    """Test the given `params`, returning the corresponding `config` on failure
+    and `None` on success or inapplicability"""
+    result = await testConfig(config, options)
+    if not options.quiet:
+        print(f"{result.name}: {config!r}")
+    if result == TestResult.FAIL:
+        return config
+    return result
+
+async def sweepParameters(paramIter: Iterable[IterType],
+        toConfig: Callable[[IterType, Options], MLIROnlyConfig],
+        options: Options) -> Tuple[int, int, List[MLIROnlyConfig]]:
+    failingConfigs = []
+    passed = 0
+    invalid = 0
+    configs = (c for c in (toConfig(p, options) for p in paramIter))
+    for configs in grouper((dropGoodConfig(c, options) for c in configs),
+            options.concurrent_tests):
+        configsFuture = asyncio.gather(*configs)
+        try:
+            configsResults = await configsFuture
+        except Exception as e:
+            configsFuture.cancel()
+            raise e
+        for result in configsResults:
+            if result == TestResult.PASS:
+                passed = passed + 1
+            elif result == TestResult.INVALID:
+                invalid = invalid + 1
+            else:
+                failingConfigs.append(result)
+
+    return (passed, invalid, failingConfigs)
+
+CONV_STRUCTURE = itertools.product(
+    # Small/large - that is, do we have padding
+    [False, True],
+    # op
+    ['fwd', 'wrw', 'bwd'],
+    # layout
+    ['NCHW', 'NHWC'],
+    # dtype
+    # TODO(kdrewnia): add bf16 once we're confident in that support
+    # and add int8 for fwd only
+    ['f32', 'f16'],
+    # Padding - hl, hr, wl, wr in [0, 3]
+    # [0, 3] hits the cases 0, < y/x, == y/x, > y/x
+    range(0, 4),
+    range(0, 4),
+    range(0, 4),
+    range(0, 4),
+    # Stride - 1 or 2 - all meaningful strides before breaking past h/w=4
+    range(1, 3),
+    range(1, 3),
+    # Dilation in [1, 2] - all meaningful dilations befor breaking past h/w=4
+    range(1, 3),
+    range(1, 3))
+
+def to_conv_structure_type_test(params, options: Options) -> MLIROnlyConfig:
+    size, op, layout, dtype, phl, phr, pwl, pwr, sh, sw, dh, dw = params
+    # Fixed parameters, y = x = 2, hi = wi = 4, g = 1
+    g, hi, wi, y, x = 1, 4, 4, 2, 2
+    if size:
+        # Values of n, c, k that prevent hitting the padding kernel
+        n, c, k = 64, 64, 64
+    else:
+        # Values of n, c, k, meant to be small and to hit the padding kernel
+        n, c, k = 1, 7, 7
```

Comment:
I'm want to seek for some clarifications:
 - `g: 1`: this makes group convolution to be always hard-coded to 1 therefore making it untested?
 - `n, c, k = 64, 64, 64`: So the reason you pick those is due to they are larger than `1,7,7` and will always processed by non-padding kernel? But no particular reason on why you don't pick another set of parameters like `128,128,128`?

I'm okay with we go with the hard-coded numbers, but I need to understand that the parameter sweeps excluded the `g, hi, wi, y, x, n, c, k` parameters?

Reply:
Yeah, I'm only sweeping two particular sets of values for g, n, w, h, y, x, c, and k . I could throw together another set of tests that sweeps that set, probably carefully picking some other sets of those values but keeping things like stride and dilation constant. 

And yeah, all-64 was the smallest input I could think of that would ensure there was no need for GEMM padding - since bigger inputs lead to more verification time.

Reply:
Two sets are a reasonable start. I think we can go with it unless there's a counter example from a different work load. Could you actually file a follow-up ticket with ROCm x.x just so that we can grab the ticket when there's a need to extend?

Reply:
No problem, will do